server/src/test/java/org/opensearch/action/support/IndicesOptionsTests.java

```diff
@@ -66,34 +66,39 @@ public void testSerialization() throws Exception {
-            indicesOptions.writeIndicesOptions(output);
-
-            StreamInput streamInput = output.bytes().streamInput();
-            streamInput.setVersion(version);
-            IndicesOptions indicesOptions2 = IndicesOptions.readIndicesOptions(streamInput);
-
-            assertThat(indicesOptions2.ignoreUnavailable(), equalTo(indicesOptions.ignoreUnavailable()));
-            assertThat(indicesOptions2.allowNoIndices(), equalTo(indicesOptions.allowNoIndices()));
-            assertThat(indicesOptions2.expandWildcardsOpen(), equalTo(indicesOptions.expandWildcardsOpen()));
-            assertThat(indicesOptions2.expandWildcardsClosed(), equalTo(indicesOptions.expandWildcardsClosed()));
-            assertThat(indicesOptions2.expandWildcardsHidden(), equalTo(indicesOptions.expandWildcardsHidden()));
-            assertThat(indicesOptions2.forbidClosedIndices(), equalTo(indicesOptions.forbidClosedIndices()));
-            assertThat(indicesOptions2.allowAliasesToMultipleIndices(), equalTo(indicesOptions.allowAliasesToMultipleIndices()));
-            assertEquals(indicesOptions2.ignoreAliases(), indicesOptions.ignoreAliases());
+            try {
+                IndicesOptions indicesOptions = IndicesOptions.fromOptions(
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean(),
+                    randomBoolean()
+                );
+
+                BytesStreamOutput output = new BytesStreamOutput();
+                output.setVersion(version);
+                indicesOptions.writeIndicesOptions(output);
+
+                StreamInput streamInput = output.bytes().streamInput();
+                streamInput.setVersion(version);
+                IndicesOptions indicesOptions2 = IndicesOptions.readIndicesOptions(streamInput);
+
+                assertThat(indicesOptions2.ignoreUnavailable(), equalTo(indicesOptions.ignoreUnavailable()));
+                assertThat(indicesOptions2.allowNoIndices(), equalTo(indicesOptions.allowNoIndices()));
+                assertThat(indicesOptions2.expandWildcardsOpen(), equalTo(indicesOptions.expandWildcardsOpen()));
+                assertThat(indicesOptions2.expandWildcardsClosed(), equalTo(indicesOptions.expandWildcardsClosed()));
+                assertThat(indicesOptions2.expandWildcardsHidden(), equalTo(indicesOptions.expandWildcardsHidden()));
+                assertThat(indicesOptions2.forbidClosedIndices(), equalTo(indicesOptions.forbidClosedIndices()));
+                assertThat(indicesOptions2.allowAliasesToMultipleIndices(), equalTo(indicesOptions.allowAliasesToMultipleIndices()));
+                assertEquals(indicesOptions2.ignoreAliases(), indicesOptions.ignoreAliases());
+            } catch (IllegalArgumentException iae) {
```

Comment:
Can we also please add a test specifically for when `expandToClosedIndices` and `forbidClosedIndices` are both true and use  `expectThrows`?  
```java
expectThrows(IllegalArgumentException.class, () -> IndicesOptions.fromOptions...
```

In these other tests we can add the assertion that when the exception is caught that those flags are both set as true or ensure they never throw by conditionally setting the flags.
```java
                final boolean expandToClosedIndices = randomBoolean();
                IndicesOptions indicesOptions = IndicesOptions.fromOptions(
                    randomBoolean(),
                    randomBoolean(),
                    randomBoolean(),
                    expandToClosedIndices,
                    randomBoolean(),
                    randomBoolean(),
                    expandToClosedIndices ? false : randomBoolean(),
                    randomBoolean(),
                    randomBoolean()
                );
```

Reply:
@mch2 Added the test and tagged you below.